lib/base-compiler.js

```diff
@@ -55,6 +56,11 @@ Compile.prototype.newTempDir = function () {
 Compile.prototype.writeFile = Promise.denodeify(fs.writeFile);
 Compile.prototype.readFile = Promise.denodeify(fs.readFile);
 Compile.prototype.stat = Promise.denodeify(fs.stat);
+Compile.prototype.couldHaveOptOutput = function(options, version) {
+    //TODO(jared): a compiler meta data gatherer could improve this
+    return version.toLowerCase().indexOf("clang") > -1 && 
+          options.some((x) => x == "-fsave-optimization-record");
```

Comment:
it's also the case that for GCC I have a list of all supported compiler options parsed out (via https://github.com/mattgodbolt/compiler-explorer/blob/master/lib/base-compiler.js#L304 )... if it's trivial to do the same then seeing if the compiler supports `-fsave-optimization-record` might be enough?